### PR TITLE
man-db: add gitlab repository, new version 2.10.1 (fixes MAXDIRS issue)

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

**Comment by haampie** [comment]
Thanks for the PR, but it seems you're trying to fix a problem that is solved in later versions of man-db.

Can you update the package?

1. They moved to gitlab: https://git.savannah.gnu.org/cgit/man-db.git/diff/MOVED_TO_GITLAB?id=5b273ae106744bbe93341d6ad3aac1980ac97b51
2. Your issue is fixed https://gitlab.com/cjwatson/man-db/-/commit/199a6e80e7b4baa27e195f2c1120c45df9ea4e67 in version 2.8.6 and later.

See also https://gitlab.com/cjwatson/man-db/-/blob/2.10.1/NEWS.md

**Comment by vvolkl** [comment]
@haampie Nice catch, it absolutely did not occur to me to check for updates on such an ancient tool. Good to see it is being actively  developed.

I didn't manage to get the autoreconf step working with the release tarball from gitlab, but using the pre-prepared `configure` from the download page it works fine.

File: var/spack/repos/builtin/packages/man-db/package.py [hunk 1]
```diff
@@ -13,14 +13,16 @@ class ManDb(AutotoolsPackage):
...
-    version('2.7.6.1', sha256='dd913662e341fc01e6721878b6cbe1001886cc3bfa6632b095937bba3238c779')
+    version('2.10.1', sha256='2ffd8f2e80122fe72e60c740c851e6a3e15c9a7921185eb4752c1c672824bed6')
+    version('2.7.6.1', sha256='08edbc52f24aca3eebac429b5444efd48b9b90b9b84ca0ed5507e5c13ed10f3f')
 
-    depends_on('autoconf')
-    depends_on('automake')
+    depends_on('pkgconf', type='build')
     depends_on('gettext')
     depends_on('libpipeline')
+    depends_on('libpipeline@1.5.0:', when='@2.10.1')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/man-db/package.py]:
```suggestion
    depends_on('libpipeline@1.5.0:', when='@2.8.0:')
    depends_on('libpipeline@1.4.0:', when='@2.7.1:')
    depends_on('libpipeline@1.3.0:', when='@2.6.7:')
    depends_on('libpipeline@1.1.0:', when='@2.6.0:')
```

**Review by haampie — CHANGES_REQUESTED** [review]
checked with `git blame configure.ac | grep PKG_CHECK_MODULES` + `git describe --contains <commit>` + `git checkout <commit>~1` a few versions back for the bounds

File: var/spack/repos/builtin/packages/man-db/package.py [hunk 2]
```diff
@@ -13,14 +13,16 @@ class ManDb(AutotoolsPackage):
...
 
-    version('2.7.6.1', sha256='dd913662e341fc01e6721878b6cbe1001886cc3bfa6632b095937bba3238c779')
+    version('2.10.1', sha256='2ffd8f2e80122fe72e60c740c851e6a3e15c9a7921185eb4752c1c672824bed6')
+    version('2.7.6.1', sha256='08edbc52f24aca3eebac429b5444efd48b9b90b9b84ca0ed5507e5c13ed10f3f')
 
-    depends_on('autoconf')
-    depends_on('automake')
+    depends_on('pkgconf', type='build')
     depends_on('gettext')
     depends_on('libpipeline')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/man-db/package.py]:
```suggestion
```

**Comment by haampie** [comment]
looks good. i'm confused why the hell `spack spec man-db` gives me 150+ packages to install though...

**Review by haampie — APPROVED** [review]

**Comment by vvolkl** [comment]
Yes, I've been battling defaults that lead to such bloat: https://github.com/spack/spack/pull/29373 should help.

**Pull request merged**